e2e/common_steps.go

```diff
@@ -223,3 +271,40 @@ func (issCtx *IssuerContext) verifyCertificateRequestState(ctx context.Context,
 
 	return nil
 }
+
+func (issCtx *IssuerContext) verifyCertificateContent(ctx context.Context, usage string) error {
+	secretName := issCtx.certName + secretSuffix
+
+	certBytes, err := getCertificateData(ctx, testContext.clientset, issCtx.namespace, secretName)
+	if err != nil {
+		assert.FailNow(godog.T(ctx), "Failed to get certificate data: "+err.Error())
+	}
+
+	if len(certBytes) == 0 {
+		assert.FailNow(godog.T(ctx), "Certificate data is empty")
+	}
+
+	decodedData, _ := pem.Decode(certBytes)
+	if decodedData == nil {
+		assert.FailNow(godog.T(ctx), "Failed to decode certificate data")
+	}
+
+	cert, err := x509.ParseCertificate(decodedData.Bytes)
+	if err != nil {
+		assert.FailNow(godog.T(ctx), "Failed to parse certificate: "+err.Error())
+	}
+
+	for _, expectedUsage := range strings.Split(usage, ",") {
```

Comment:
What about the rest of the content? This function is only validating keyUsages right?